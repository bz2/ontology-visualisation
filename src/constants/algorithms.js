export const ALGO_TYPE_FULL = 'full'
export const ALGO_TYPE_NEIGHBOURHOOD = 'neighbourhood'
export const ALGO_TYPE_SHORTEST_PATH = 'shortest-path'
export const ALGO_TYPE_BOUNDING_BOX = 'bounding-box'
export const ALGO_TYPE_NODES_FILTER = 'nodes-filter'
export const ALGO_TYPE_EDGES_FILTER = 'edges-filter'
