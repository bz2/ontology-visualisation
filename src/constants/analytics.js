export const USERBACK_ACCESS_TOKEN = '29230|42198|CVJlhtfkaYN3J3xP2x200ghJ1'
export const USERBACK_COOKIE = 'ubwc'
export const USERBACK_WIDGET = 'https://static.userback.io/widget/v1.js'
export const GA_ID = 'UA-195025859-1'
