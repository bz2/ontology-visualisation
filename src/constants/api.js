export const API_ENDPOINT = `${process.env.endpointUrl}/api` // 'http://localhost:8080/api'

export const GET_ONTOLOGY = `${API_ENDPOINT}/ontology`
export const GET_GRAPH = `${API_ENDPOINT}/graph`
export const GET_GRAPH_QUERY = `${GET_GRAPH}/query`

export const GET_EDGES = `${GET_GRAPH}/edges`
export const GET_EDGE = `${GET_GRAPH}/edges/{id}`

export const GET_NODES = `${GET_GRAPH}/nodes`
export const GET_NODE = `${GET_GRAPH}/nodes/{id}`

// export const GET_PROPERTIES = `${GET_GRAPH}/properties`
// export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
// export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
// export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
// export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`
export const GET_PROPERTIES = `${GET_ONTOLOGY}/properties`
export const GET_PROPERTIES_OBJECTS = `${GET_PROPERTIES}/objects`
export const GET_PROPERTIES_OBJECT = `${GET_PROPERTIES}/objects/{id}`
export const GET_PROPERTIES_ANNOTATIONS = `${GET_PROPERTIES}/annotations`
export const GET_PROPERTIES_ANNOTATION = `${GET_PROPERTIES}/annotations/{id}`

export const AUTH_SIGN_IN = `${API_ENDPOINT}/login`
