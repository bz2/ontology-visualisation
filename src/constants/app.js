export const APP_NAME = 'OntoPop'
