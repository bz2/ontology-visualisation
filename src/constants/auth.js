export const AUTH_COOKIE = 'ontology_viz_auth'
