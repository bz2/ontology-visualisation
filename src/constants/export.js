export const EXPORT_JPG = 'jpeg'
export const EXPORT_PNG = 'png'
export const EXPORT_PDF = 'pdf'
export const EXPORT_CSV = 'csv'
export const EXPORT_OWL = 'owl'

export const EXPORT_GRAPH_OPTIONS = [
  EXPORT_JPG,
  EXPORT_PNG,
  EXPORT_PDF
]

export const EXPORT_DATA_OPTIONS = [
  EXPORT_CSV,
  EXPORT_OWL
]
