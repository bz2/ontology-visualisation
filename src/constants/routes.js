export const ROUTE_INDEX = '/'
export const ROUTE_LOGIN = '/login'
export const ROUTE_PROFILE = '/profile'
export const ROUTE_FORGOT_PASSWORD = '/forgot-password'
export const ROUTE_SEARCH = '/search'
export const ROUTE_NETWORK_GRAPHS = '/network-graphs'
export const ROUTE_NETWORK_GRAPH_OPTIONS = '/network-graph-options'
export const ROUTE_FREE_TEXT_SEARCH = '/free-text-search'
export const ROUTE_STRUCTURED_SEARCH = '/structured-search'
export const ROUTE_NODES_SELECTION = '/nodes-selection'
export const ROUTE_EDGES_SELECTION = '/edges-selection'
export const ROUTE_NODES_FILTER = '/nodes-filter'
export const ROUTE_EDGES_FILTER = '/edges-filter'
export const ROUTE_BOUNDING_BOX = '/bounding-box'
export const ROUTE_NODE_NEIGHBOURHOOD = '/node-neighbourhood'
export const ROUTE_SHORTEST_PATH = '/shortest-path'
export const ROUTE_SETTINGS = '/settings'
export const ROUTE_CUSTOM_QUERY = '/custom-query'
export const ROUTE_STYLING = '/styling'
export const ROUTE_NOTES = '/notes'
export const ROUTE_SYNONYMS = '/synonyms'
export const ROUTE_EXPORT = '/export'
export const ROUTE_EDIT_ONTOLOGY = '/edit-ontology'

export const AUTH_ROUTES = [
  ROUTE_LOGIN,
  ROUTE_FORGOT_PASSWORD
]

export const FORM_LINK = 'https://forms.office.com/Pages/ResponsePage.aspx?id=sp9QKa9_i0-3ojL5bsXebMHGED18T-VBrbTcXx4rIrtUNVdHRDQ1SVdQVFg4UDlTR1MwWThITjM4Qy4u'
