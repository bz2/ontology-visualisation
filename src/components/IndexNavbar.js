/* eslint max-len:0 */
import Link from 'next/link'
import { useTranslation } from 'react-i18next'
import { useRouter } from 'next/router'
import logo from '../assets/images/logo.png'
import { APP_NAME } from '../constants/app'
import { NAVBAR } from '../constants/homepage'
import { ROUTE_INDEX } from '../constants/routes'

const IndexNavbar = () => {
  const { t } = useTranslation()
  const router = useRouter()

  return (
    <>
      <nav className="top-0 fixed z-50 w-full flex flex-wrap items-center justify-between navbar-expand-lg bg-white shadow website-navbar">
        <div className="container mx-auto flex items-center justify-between">
          <Link href={`${ROUTE_INDEX}#top`}>
            <a
              className="text-blueGray-700 text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase"
            >
              <img
                src={logo}
                alt={APP_NAME}
                className="h-16"
              />
            </a>
          </Link>

          <div
            className="website-navbar-items flex"
          >
            {
              NAVBAR.map((nav) => (
                <Link
                  key={`nav-${nav}`}
                  href={`${ROUTE_INDEX}#${nav}`}
                >
                  <a
                    className={`p-link ${router.asPath.includes(nav) ? 'p-link-selected' : ''}`}
                  >
                    {t(nav)}
                  </a>
                </Link>
              ))
            }
          </div>
        </div>
      </nav>
    </>
  )
}

export default IndexNavbar
