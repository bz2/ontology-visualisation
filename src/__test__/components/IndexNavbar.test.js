import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { useRouter } from 'next/router'
import IndexNavbar from '../../components/IndexNavbar'
import { ROUTE_INDEX } from '../../constants/routes'

jest.mock('next/router')

const setup = () => {
  const props = { }

  const component = shallow(<IndexNavbar {...props} />)

  return {
    component,
    props
  }
}

describe('IndexNavbar', () => {
  it('should match snapshot', () => {
    useRouter.mockImplementation(() => ({
      asPath: ROUTE_INDEX
    }))

    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
