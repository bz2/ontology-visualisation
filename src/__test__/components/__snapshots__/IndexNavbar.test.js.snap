// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexNavbar should match snapshot 1`] = `
<Fragment>
  <nav
    className="top-0 fixed z-50 w-full flex flex-wrap items-center justify-between navbar-expand-lg bg-white shadow website-navbar"
  >
    <div
      className="container mx-auto flex items-center justify-between"
    >
      <Link
        href="/#top"
      >
        <a
          className="text-blueGray-700 text-sm font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap uppercase"
        >
          <img
            alt="OntoPop"
            className="h-16"
            src={
              Object {
                "process": [Function],
              }
            }
          />
        </a>
      </Link>
      <div
        className="website-navbar-items flex"
      >
        <Link
          href="/#about"
          key="nav-about"
        >
          <a
            className="p-link "
          >
            about
          </a>
        </Link>
        <Link
          href="/#features"
          key="nav-features"
        >
          <a
            className="p-link "
          >
            features
          </a>
        </Link>
        <Link
          href="/#app"
          key="nav-app"
        >
          <a
            className="p-link "
          >
            app
          </a>
        </Link>
        <Link
          href="/#contribute"
          key="nav-contribute"
        >
          <a
            className="p-link "
          >
            contribute
          </a>
        </Link>
        <Link
          href="/#contacts"
          key="nav-contacts"
        >
          <a
            className="p-link "
          >
            contacts
          </a>
        </Link>
      </div>
    </div>
  </nav>
</Fragment>
`;
